packages/react/src/components/MultiSelect/FilterableMultiSelect.js

```diff
@@ -200,7 +200,10 @@ const FilterableMultiSelect = React.forwardRef(function FilterableMultiSelect(
   }
 
   function clearInputValue() {
-    setInputValue('');
+    textInput.current.value.length === 1
+      ? setInputValue('')
+      : setInputValue(textInput.current.value);
+
```

Comment:
I think we need to check if the key pressed is <kbd>esc</kbd>. If so, it should clear the input (if the menu is closed). It does close the menu on the first press (expected), but I can't clear the input anymore.